packages/redeem_solver/src/service.ts

```diff
@@ -83,6 +95,47 @@ export class RedeemService {
 			);
 		}
 	}
+
+	private async finalizeRequest(req: RedeemRequest) {
+		const now = Date.now();
+		const deadline = req.created_at + this.redeemDurationMs;
+
+		if (now <= deadline) {
+			// Not yet ready to finalize
+			return;
+		}
+
+		logger.info({
+			msg: "Finalizing redeem request",
+			redeemId: req.redeem_id,
+		});
+
+		try {
+			const client = this.getSuiClient(req.sui_network);
```

Comment:
add a comment that we don't want to ptb here

Reply:
done